azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -36,20 +37,24 @@
 TRANSPORT_PRV_FILE_NAME = "TransportPrivate.pem"
 
 
+class GoalStateRetrievalMode:
+    Default = 0
+    Limited = 1
+    Full = 2
+
+
 class GoalState(object):
     #
     # Some modules (e.g. telemetry) require an up-to-date container ID. We update this variable each time we
     # fetch the goal state.
     #
     ContainerID = "00000000-0000-0000-0000-000000000000"
+    _HttpFailedIndicator = "[HTTP Failed]"
 
-    def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
+    def __init__(self, wire_client, ext_config_retriever, retrieval_mode=GoalStateRetrievalMode.Default):
```

Comment:
ext_config_retriever is not needed as parameter since we are already passing the wire client

Reply:
same comment for the fetch*goal_state methods

Reply:
"GoalStateRetrievalMode.Default" is not very descriptive, can we rename it to "GoalStateRetrievalMode.GoalStateChanged" or something like that?

Reply:
Removed ext_config_retriever and renamed Default to GoalStateChanged